pkg/vcdsdk/gateway.go

```diff
@@ -2209,3 +2224,51 @@ func (gm *GatewayManager) reserveIpForLoadBalancer(ctx context.Context, claimMar
 	// Was unable to reserve an Ip on any of the available Ip Spaces
 	return "", fmt.Errorf("unable to reserve Ip from any available Ip spaces")
 }
+
+// releaseIpFromLoadBalancer will scan through all Ip Spaces available to the gateway for an existing allocation
+// (description of allocation will contain cluster id, service name and namespace). If such an allocation can't be
+// retrieved, the method will return without raising any error. It should be assumed that the allocation was removed in
+// a previous attempt. If an allocation is found, it be deleted, thereby releasing the IP from the load balancer as well
```

Comment:
nit: "If an allocation is found, it be deleted" -> "If an allocation is found, it **should** be deleted"